openassessment/xblock/static/js/src/lms/oa_grade.js

```diff
@@ -20,17 +22,31 @@ OpenAssessment.GradeView.prototype = {
         var view = this;
         var baseView = this.baseView;
         var stepID = ".step--grade";
+        view.isRendering = true;
         this.server.render('grade').done(
             function(html) {
                 // Load the HTML and install event handlers
                 $(stepID, view.element).replaceWith(html);
                 view.server.renderLatex($(stepID, view.element));
+                view.isRendering = false;
                 view.installHandlers();
                 if (typeof usageID !== 'undefined' &&
                     !$(stepID, view.element).hasClass("is--unfinished") &&
                     !$(stepID, view.element).hasClass("is--unstarted") &&
                     !$(stepID, view.element).hasClass("is--waiting--staff")) {
                     $("[id='oa_grade_" + usageID + "']", view.element).focus();
+                    view.announceStatusChange = true;
+                }
+                if (typeof usageID !== 'undefined' &&
+                    $(stepID, view.element).hasClass("is--showing")){
+                    view.announceStatusChange = true;
+                }
+                if (view.announceStatusChange) {
+                    // Update the SR and announce the status change
+                    view.baseView.announceStatusChangeToSR(view.baseView.getAssessmentGradeStatus(stepID, usageID));
+                    view.announceStatusChange = false;
+                } else {
+                    view.baseView.announceStatusChangeToSR();
```

Comment:
The views can complete their render in a different order, this method without parameters checks to see if the views are still rendering and if they are done adds the text to the aria live region.